### DISCUSSION TITLE
Can we Remove the Project Validator for Project Emails?

### DISCUSSION THREAD

**Discussion by aljawaid** [kanboard/kanboard#5126] [Q&A] [answered]
https://github.com/kanboard/kanboard/blob/5c25755f9e43b6b3ab85a5bc68d7ab4afa5e3534/app/Validator/ProjectValidator.php#L38

This commit was done 6 years ago, I think it needs changing. I am in a situation where the same email address is being used for different projects but this validator stops me from getting email reports.

Can this line be removed safely?  Or can somebody help/guide how adjust the code to allow for the same email address to be used across multiple projects?

**Comment by fguillot** [comment, marked as answer]
The original commit is there: https://github.com/kanboard/kanboard/commit/d5c4c18ea0bd6c0d1a9d0104c1c17748a35f85a7

Removing the unique constraint would lead to unpredictable behavior when fetching a project by email from the database.

https://github.com/kanboard/kanboard/blob/5c25755f9e43b6b3ab85a5bc68d7ab4afa5e3534/app/Model/ProjectModel.php#L139-L146
